aodncore/util/misc.py

```diff
@@ -95,6 +96,24 @@ def format_exception(exception):
     return "{cls}: {message}".format(cls=exception.__class__.__name__, message=exception)
 
 
+def get_fields_from_filename(filepath, pattern):
+    """Function to retrieve parts of a filename given a compiled pattern (re.compile(str))
+    :return: dictionary of fields matching a given pattern
+    """
+    retype = type(re.compile(''))
+    if not isinstance(pattern, retype):
+        try:
+            pattern = re.compile(pattern)
+        except Exception as err:
+            raise TypeError("pattern: \"{pattern}\" could not be compiled as a \"{retype}\" instance".format(
```

Comment:
I don't really like escaping quotes, so usually use single quotes *inside* the string. Also, I think the message can be simpler, since the type stuff isn't really relevant to know. What is relevant is what the internal error was, so I'd change it to this:

```python
"pattern: \"{pattern}\" could not be compiled as a \"{retype}\" instance"
```
```python
"error compiling pattern '{pattern}': {e}".format(pattern=pattern, e=format_exception(e))
```